insta/src/snapshot.rs

```diff
@@ -540,7 +538,10 @@ impl Snapshot {
                     TextSnapshotKind::Inline => contents_match_exact,
                 }
             }
-            (SnapshotContents::Binary(a), SnapshotContents::Binary(b)) => a == b,
+            (SnapshotContents::Binary(a), SnapshotContents::Binary(b)) => {
+                // For binary snapshots the extension also need to be the same:
+                a == b && self.metadata.snapshot_type == other.metadata.snapshot_type
+            }
```

Comment:
tiny nit: we could just call `.matches` in `.matches_fully` and encapsulate this there